### DISCUSSION TITLE
Rendering in two windows?

### DISCUSSION THREAD

**Discussion by abailey20** [bonsai-rx/bonsai#1842] [Q&A]
Is it theoretically possible to render two different visual stimuli in two different windows? For example, creating a converging stimulus using two grating moving towards each other on two different screens.

I see no way to assign which stimuli is played in which window created.

Has anyone managed to do this or find a way to create the same effect?

Many thanks!

**Comment by RoboDoig** [comment]
It depends on your screen setup, but one approach is to create a shader window that covers the full extent of both screens. You can then use `DrawViewport` to restrict rendering to a portion of the window to draw to the different screens.

**Reply by abailey20** [reply]
Hi @RoboDoig

I I have tried to apply what you have suggested but can't seem to get around 'No method overload found for the given arguments' - I am not sure what this means but think it's something to do with correctly assigning the gratings to different parts of the windows?

In the file I have attached I attempted to create a window 200x400pix and assign two different gratings to each half of the screen so that each is 200x200pix.

Can you see where I've gone wrong? 
Thanks for your response. 

[MosaicScreenGratings.zip](https://github.com/user-attachments/files/15871244/MosaicScreenGratings.zip)
